vhdbuilder/packer/convert-sig-to-classic-storage-account-blob.sh

```diff
@@ -83,13 +83,21 @@ echo "Granting access to $disk_resource_id for 1 hour"
 # shellcheck disable=SC2102
 sas=$(az disk grant-access --ids $disk_resource_id --duration-in-seconds 3600 --query [accessSas] -o tsv)
 
-# TBD: Need to investigate why `azcopy-preview login --login-type=MSI` does not work
-echo "Setting azcopy environment variables with pool identity: $AZURE_MSI_RESOURCE_STRING"
-export AZCOPY_AUTO_LOGIN_TYPE="MSI"
-export AZCOPY_MSI_RESOURCE_STRING="$AZURE_MSI_RESOURCE_STRING"
-
 echo "Uploading $disk_resource_id to ${CLASSIC_BLOB}/${CAPTURED_SIG_VERSION}.vhd"
-azcopy-preview copy "${sas}" "${CLASSIC_BLOB}/${CAPTURED_SIG_VERSION}.vhd" --recursive=true || exit $?
+
+if [ "${OS_TYPE,,}" == "linux" ]; then
+  # Use azcopy login for Linux
+  azcopy login --login-type=MSI
```

Comment:
is this something we have to do? can we just keep using the current approach for both OS types?

Reply:
Removing the go dev container and using this login method was the only way I could get the convert to vhd blob step to work with Ubuntu. I messaged you on teams about my last ditch effort to continue using the container. I added the os check so that windows workflow was unchanged and they wouldnt end up with failures from my changes since their pipeline uses a container as well.

This was an issue a couple months ago too, but changing the go dev container to version 1.38.0 fixed it. Now it doesn't
help for some reason :/